app/plugins/enabled.js

```diff
@@ -0,0 +1,15 @@
+const { endpointEnabled } = require('../config')
+
+module.exports = {
+  plugin: {
+    name: 'enabled',
+    register: (server) => {
+      server.ext('onRequest', (request, h) => {
+        if (!endpointEnabled && request.path.startsWith('/statement/')) {
+          return h.response('Service is intentionally disabled in this environment').code(503).takeover()
+        }
+        return h.continue
```

Comment:
Hooking this into the Hapi lifecycle, any request to a `/statement/` route will be rejected if the service is disabled.

It's important to allow other routes still as k8s health probes access the `/healthy` and `/healthz` routes otherwise the pod would be continuously reported as "dead" and would be forever restarted.